reana_db/utils.py

```diff
@@ -256,7 +258,7 @@ def store_workflow_disk_quota(workflow):
         disk_bytes = 0
     if workflow_resource:
         workflow_resource.quantity_used = disk_bytes
-    else:
+    elif inspect(workflow).persistent:
```

Comment:
Question: what is the root problem that leads us to do this change? Is it that the workflow is not yet committed and we do `Session.add(workflow_resource)` and the workflow does not exist yet on DB?
It looks a bit hacky IMO, but if there's no other way...

BTW, now that there's an `elif`, if none of these checks pass, we'd be doing a `Session.commit()` in any case, is that correct?

Reply:
> Question: what is the root problem that leads us to do this change? Is it that the workflow is not yet committed and we do Session.add(workflow_resource) and the workflow does not exist yet on DB?
It looks a bit hacky IMO, but if there's no other way...

I completely agree, this is a hack. The root cause is that, using [the status change listener approach](https://github.com/reanahub/reana-db/blob/c03aa75fed2b0c5d3aa4cc3555bf72ac3a3cf6b0/reana_db/models.py#L579-L580) might not be the best approach to handle these situations because the status changes [when the SQLAlchemy object that represents the workflow is created in memory (in the constructor)](https://github.com/reanahub/reana-db/blob/c03aa75fed2b0c5d3aa4cc3555bf72ac3a3cf6b0/reana_db/models.py#L432), hence, triggering the listener when the workflow is not yet in DB. It happened that we discovered it now because we try to make reference to the workflow table for the first time inside the listener logic. The options I was left with right now were: (1) refactoring or (2) adding this patch to get things going for now.

My fault for not correctly stating (1) and (2) in the first place and justifying that (2) is chosen because of time/blocking other issues.

@mvidalgarcia what do you think about creating an issue once this is merged so we can reference to the problem in the code?

> BTW, now that there's an elif, if none of these checks pass, we'd be doing a Session.commit() in any case, is that correct?

Thanks, good catch, I am amending it.